### PR TITLE
[DAG] Adding m_FPToUI and m_FPToSI to SDPatternMatch.h

### LINKED ISSUE #103872: [DAG] SDPatternMatch - add m_FPToUI / m_FPToSI matchers
Add m_FPToUI / m_FPToSI matchers for ISD::FP_TO_UINT / ISD::FP_TO_SINT nodes to SDPatternMatch.h and suitable test coverage to SelectionDAGPatternMatchTest.cpp

### PR DESCRIPTION
**Pull request by jf-botto**
Fixes https://github.com/llvm/llvm-project/issues/103872.

Adds m_FPToUI/m_FPToSI matchers for ISD::FP_TO_UINT/ISD::FP_TO_SINT in SDPatternMatch.h with suitable test coverage.

### CODE REVIEW

**Comment by jf-botto** [comment]
CC: @RKSimon

File: llvm/unittests/CodeGen/SelectionDAGPatternMatchTest.cpp [hunk 1]
```diff
@@ -263,6 +268,9 @@ TEST_F(SelectionDAGPatternMatchTest, matchUnaryOp) {
   EXPECT_FALSE(sd_match(Sub, m_Neg(m_Value())));
   EXPECT_FALSE(sd_match(Neg, m_Not(m_Value())));
   EXPECT_TRUE(sd_match(VScale, m_VScale(m_Value())));
+
+  EXPECT_TRUE(sd_match(FPToUI, m_FPToUI(m_Value())));
+  EXPECT_TRUE(sd_match(FPToSI, m_FPToSI(m_Value())));
```
Comment (RKSimon) [inline llvm/unittests/CodeGen/SelectionDAGPatternMatchTest.cpp:273]: Add a fail test as well, maybe matching the fptoui to FPToSi?

Comment (jf-botto) [reply llvm/unittests/CodeGen/SelectionDAGPatternMatchTest.cpp:273]: Thank you. Added.

**Review by RKSimon — APPROVED** [review]
LGTM - cheers

**Comment by jf-botto** [comment]
> LGTM - cheers

Would you mind merging it, please? I don't have write access.

**Pull request merged**